frontend/src/Frontend/UI/Dialogs/Send.hs

```diff
@@ -166,94 +192,81 @@ previewTransfer model fromAccount fromGasPayer toAddress crossChainData amount =
-      pure $ AccountRef_NonVanity key chain ==> nv
-    mKey = textToKey $ unAccountName name
     name = _kadenaAddress_accountName ka
     chain = _kadenaAddress_chainId ka
 
 -- | Perform a same chain transfer or transfer-create
 sameChainTransfer
-  :: (MonadWidget t m, HasCrypto key m, Monoid mConf, HasLogger model t)
+  :: (MonadWidget t m, HasCrypto key m, Monoid mConf, HasLogger model t, HasTransactionLogger m)
   => model
   -> SharedNetInfo NodeInfo
   -> KeyStorage key
-  -> Account
+  -> (AccountName, ChainId, Account)
   -- ^ From account
-  -> Account
+  -> (AccountName, Account)
   -- ^ Gas payer
   -> KadenaAddress
   -- ^ Recipient account
   -> Decimal
   -- ^ Amount to transfer
   -> Workflow t m (mConf, Event t ())
-sameChainTransfer model netInfo keys fromAccount gasPayer toAccount amount = Workflow $ do
-  let recipientCreated = _kadenaAddress_accountCreated toAccount
+sameChainTransfer model netInfo keys (fromName, fromChain, fromAcc) (gasPayer, gasPayerAcc) toAccount amount = Workflow $ do
+  let mKeyset = _kadenaAddress_keyset toAccount
   let code = T.unwords $
-        [ "(coin." <> case recipientCreated of
-          AccountCreated_Yes -> "transfer"
-          AccountCreated_No -> "transfer-create"
-        , tshow $ unAccountName $ accountToName fromAccount
+        [ "(coin." <> case mKeyset of
+          Nothing -> "transfer"
+          Just _ -> "transfer-create"
+        , tshow $ unAccountName fromName
         , tshow $ unAccountName $ _kadenaAddress_accountName toAccount
-        , case recipientCreated of
-          AccountCreated_Yes -> mempty
-          AccountCreated_No -> "(read-keyset 'key)"
+        , case mKeyset of
+          Nothing -> mempty
+          Just _ -> "(read-keyset 'key)"
         , tshow amount
         , ")"
         ]
-      signingSet = Set.fromList [accountKey fromAccount, accountKey gasPayer]
+      signingSet = Set.unions [accountKeys fromAcc, accountKeys gasPayerAcc]
       signingPairs = filterKeyPairs signingSet keys
       transferCap = SigCapability
         { _scName = QualifiedName { _qnQual = "coin", _qnName = "TRANSFER", _qnInfo = def }
         , _scArgs =
-          [ PLiteral $ LString $ unAccountName $ accountToName fromAccount
+          [ PLiteral $ LString $ unAccountName fromName
           , PLiteral $ LString $ unAccountName $ _kadenaAddress_accountName toAccount
           , PLiteral $ LDecimal amount
           ]
         }
-      dat = case _kadenaAddress_accountCreated toAccount of
-        AccountCreated_No
-          -- This makes the assumption that the only non-created accounts are
-          -- non-vanity: that is, the name is the public key.
-          | Right pk <- parsePublicKey (unAccountName $ _kadenaAddress_accountName toAccount)
-          -> HM.singleton "key" $ Aeson.toJSON $ KeySet (Set.singleton $ toPactPublicKey pk) (Name $ BareName "keys-all" def)
+      dat = case mKeyset of
+        -- TODO check against chain
```

Comment:
Does this block the PR?

Reply:
No, I think any of these "check against chain / lookup guard" things are just nice to have. We already will have looked it up at some point (hopefully fairly recently), but it'd be nice to ensure we are up to date.